mmv1/third_party/terraform/tests/resource_dataflow_flex_template_job_test.go.erb

```diff
@@ -114,31 +125,331 @@ func TestAccDataflowFlexTemplateJob_withServiceAccount(t *testing.T) {
 	randStr := randString(t, 10)
 	job := "tf-test-dataflow-job-" + randStr
 	accountId := "tf-test-dataflow-sa" + randStr
-	zone := "us-central1-b"
+	accountId2 := "tf-test-dataflow-sa2" + randStr
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDataflowJobDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataflowFlexTemplateJob_serviceAccount(job, accountId),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataflowJobExists(t, "google_dataflow_flex_template_job.job"),
+					testAccDataflowFlexTemplateJobHasServiceAccount(t, "google_dataflow_flex_template_job.job", accountId),
+				),
+			},
+			{
+				ResourceName:            "google_dataflow_flex_template_job.job",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"on_delete", "parameters", "skip_wait_on_job_termination", "state"},
+			},
+			{
+				Config: testAccDataflowFlexTemplateJob_serviceAccount(job, accountId),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataflowJobExists(t, "google_dataflow_flex_template_job.job"),
+					testAccDataflowFlexTemplateJobHasServiceAccount(t, "google_dataflow_flex_template_job.job", accountId2),
+				),
+			},
+			{
+				ResourceName:            "google_dataflow_flex_template_job.job",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"on_delete", "parameters", "skip_wait_on_job_termination", "state"},
+			},
+		},
+	})
+}
+
+func TestAccDataflowFlexTemplateJob_withMachineTypeCreateAndUpdate(t *testing.T) {
```

Comment:
Could we combine these tests where possible? Many of the fields look like they could be included in one config, and tested in one step. Also, per my comment from a few weeks ago, now that you've made the resource importable, you can likely rely on `ImportStateVerify` and get rid of the `Check`s.

Reply:
hi @roaks3 , could you please clarify on this. are you suggesting we should remove individual tests for update dataflow jobs for each option to a single update function with all parameters?

Regards
Sai

Reply:
Yes, exactly, where it is possible (for example, you might need to test `network` and `subnetwork` with separate tests).

For an example, you can see [`TestAccCloudRunV2Service_cloudrunv2ServiceFullUpdate`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/tests/resource_cloud_run_v2_service_test.go#L9). This test creates an initial resource with as many fields as possible in one request, and then updates as many as it can in the next step. Then, following the full test, there are a number of more focused tests that could not fit into the "full" version.

Reply:
hi @roaks3,

I have merged the tests that support update as FullUpdate where I believe it is possible to merge them.

Regards
Sai

Reply:
hi @roaks3, Is this good to merge? or did I miss anything from your review comments?

Reply:
This looks great, thanks!